wrappers/python/indy_vdr/resolver.py

```diff
@@ -1,19 +1,132 @@
+        """Remove a ledger from the resolver."""
+        try:
+            del self.pool_map[namespace]
+        except KeyError:
+            raise VdrError(VdrErrorCode.WRAPPER, f"Ledger {namespace} not configured")
+
+    async def resolve(self, did: str) -> Dict:
         """Resolve a DID to retrieve a DID Doc."""
-        result = await bindings.resolve(self.handle, did)
+        namespace = get_namespace(did)
+        pool = self.pool_map.get(namespace)
+        if not pool:
+            if self.autopilot:
+                try:
+                    pool_map = get_genesis_txns_from_did_indy_repo_by_name([namespace])
+                    pool = await open_pool(pool_map[namespace])
+                    self.add_ledger(namespace, pool)
+                except KeyError:
+                    raise VdrError(
+                        VdrErrorCode.WRAPPER, f"Unknown DID namespace: {namespace}"
+                    )
+
+            else:
+                raise VdrError(
+                    VdrErrorCode.WRAPPER, f"Unknown DID namespace: {namespace}"
+                )
+        pool_handle = getattr(pool, "handle")
+        result = json.loads(await bindings.resolve(pool_handle, did))
+        
+        reply_data = json.loads(result["didDocumentMetadata"]["nodeResponse"]["result"]["data"])
+        diddoc_content = reply_data.get("diddocContent", None)
+
+        # Handle legacy case, where diddocContent is not present and we want to check for
+        # associated ATTRIB endpoints. We can't handle in this in libindy_vdr directly. 
+        if not diddoc_content:
+            unqualified_did = reply_data["dest"]
+            req = build_get_attrib_request(None, unqualified_did, "endpoint", None, None)
+            res = await pool.submit_request(req)
+            data = res.get("data", None)
+            if data:
+                data = json.loads(data)
+                endpoints = data.get("endpoint", None)
+
+                if endpoints:
+                    services = []
+                    for (service_type, service_endpoint) in endpoints.items():
+                        if service_type == "endpoint":
+
+                            services.append({
+                            "id": f"did:indy:{namespace}:{unqualified_did}#did-communication",
+                            "type": "did-communication",
+                            "recipientKeys": [f"did:indy:{namespace}:{unqualified_did}#verkey"] ,
+                            "routingKeys": [],
+                            "priority": 0
+                            })
+                        
+                        else:
+
+                            services.append({
+                            "id": f"did:indy:{namespace}:{unqualified_did}#{service_type}",
+                            "type": service_type,
+                            "serviceEndpoint": service_endpoint 
+
+                            })
+                    result["didDocument"]["services"] = services;
```

Comment:
After countless hours trying to handle fetching the legacy endpoint (ATTRIB tx) on the rust side before crossing the ffi bridge I resigned.

I don't like it but I have added this code to the python wrapper to fetch the ATTRIB data and attach it to the did document.

If the `PoolResolver` is used from Rust projects (e.g. the proxy server) then this custom code is not necessary, but handled internally.